src/pipe_operator.rs

```diff
@@ -0,0 +1,57 @@
+use crate::util;
+
+pub fn pipe(command: String) {
+    
+    let command = command;
+
+    if command.contains("|"){
+
+        if command.starts_with("|") {
+
+            println!("Invalid command: command starts with pipe, try modifying the command");
+
+        }
+        else if command.ends_with("|"){
+
+            println!("Invalid command: command ends with pipe, try modifying the command");
+        }
+        else{
+            
+            let chunks: Vec<_> = command.split("|").collect();
+            
+            let mut flag = 0;
+
+            for substring in chunks.iter(){
+
+                let substring = substring.trim().to_string();
+
+                if substring.chars().count() == 0{
+                    
+                    flag = 1;
+                    break
+
+                }
+
+            }
+
+            if flag == 1{
+
+                println!("Invalid command");
+
+            }
+            else{
+
+                for items in chunks.iter(){
+
+                    let cmd = items.trim().to_string();
+                    let history = util::initialize_vector();
```

Comment:
don't initialize a new vector instance here. Instead, take history as the command argument, pass history from util itself and return history from this function, This way, the ownership of variable history is again returned back to the caller

Reply:
This is a dummy history vector. It does not update the actual history vector in the main. Taking the history from the main will update the vector and would need to remove the last few commands

Reply:
This is also an immutable history vector which is initialized